storefront-reference-architecture-master/cartridges/int_external_customers_data/cartridge/controllers/Account.js

```diff
@@ -0,0 +1,403 @@
+            registrationForm.customer.emailconfirm.valid = false;
+            registrationForm.customer.emailconfirm.error = Resource.msg(
+                "error.message.mismatch.email",
+                "forms",
+                null
+            );
+            registrationForm.valid = false;
+        }
+
+        if (
+            registrationForm.login.password.value !==
+            registrationForm.login.passwordconfirm.value
+        ) {
+            registrationForm.login.password.valid = false;
+            registrationForm.login.passwordconfirm.valid = false;
+            registrationForm.login.passwordconfirm.error = Resource.msg(
+                "error.message.mismatch.password",
+                "forms",
+                null
+            );
+            registrationForm.valid = false;
+        }
+
+        if (
+            !CustomerMgr.isAcceptablePassword(
+                registrationForm.login.password.value
+            )
+        ) {
+            registrationForm.login.password.valid = false;
+            registrationForm.login.passwordconfirm.valid = false;
+            registrationForm.login.passwordconfirm.error = Resource.msg(
+                "error.message.password.constraints.not.matched",
+                "forms",
+                null
+            );
+            registrationForm.valid = false;
+        }
+
+        // setting variables for the BeforeComplete function
+        var registrationFormObj = {
+            firstName: registrationForm.customer.firstname.value,
+            lastName: registrationForm.customer.lastname.value,
+            phone: registrationForm.customer.phone.value,
+            email: registrationForm.customer.email.value,
+            emailConfirm: registrationForm.customer.emailconfirm.value,
+            password: registrationForm.login.password.value,
+            passwordConfirm: registrationForm.login.passwordconfirm.value,
+            validForm: registrationForm.valid,
+            form: registrationForm,
+        };
+
+        if (registrationForm.valid) {
+            res.setViewData(registrationFormObj);
+
+            this.on("route:BeforeComplete", function (req, res) {
+                // eslint-disable-line no-shadow
+                var Transaction = require("dw/system/Transaction");
+                var accountHelpers = require("*/cartridge/scripts/helpers/accountHelpers");
+                var authenticatedCustomer;
+                var serverError;
+
+                // getting variables for the BeforeComplete function
+                var registrationForm = res.getViewData(); // eslint-disable-line
+
+                if (registrationForm.validForm) {
+                    var login = registrationForm.email;
+                    var password = registrationForm.password;
+
+                    const uniqueExternalId = UUIDUtils.createUUID();
+
+                    var response =
+                        extarnalCustomersService.externalCustomersService({
```

Comment:
This needs to be refactored as:
- Not meeting requirement that some sites (with the same codebase) should not have this feature enabled
- The controller should not know the URL for the service.

Reply:
fixed